File: materias 3 periodo ple/programacao orientada ao objeto/trabalhos praticos/tp-1/Codigo/Cpp/Cliente.cpp
#include"../Hpp/Cliente.hpp"

Cliente :: Cliente(string nome, string cpf, string rg, string endereco, string telefone, string email) : Pessoa(nome, cpf, rg, endereco, telefone, email) {}

Cliente :: ~Cliente(){};

// Sets para edição dos campos do cliente de acordo com os dados de pessoa

void Cliente :: setNomeCliente(string nome) {
    this->setNome(nome);
}

void Cliente :: setTelefoneCliente(string telefone) {
    this->setTelefone(telefone);
}

void Cliente :: setEmailCliente(string email) {
    this->setEmail(email);
}

void Cliente :: setEnderecoCliente(string endereco) {
    this->setEndereco(endereco);
}

// Interações com a lista de carro(s)

void Cliente :: addCarro(Carro carro){  
    carros.push_back(carro);
}

void Cliente :: popCarro(int id){    

    for (std::list<Carro>::iterator i = this->carros.begin(); i!=this->carros.end(); i++){
        if(i->id == id){
            this->carros.erase(i);
        }
        else{
            continue;
        }        
    }  
}

void Cliente :: printaCarro() {
    for(list<Carro>::iterator i = this->carros.begin(); i != this->carros.end(); i++ ) {
        cout << i->getModelo() << endl;
        cout << i->getPlaca() << endl;
        cout << i->getCor() << endl;
    }
}

File: materias ple/programacao orientada ao objeto/trabalhos praticos/tp-2/SistemaOficina/src/sistemaoficina/Clientes/Cliente.java
package sistemaoficina.Clientes;
import sistemaoficina.Clientes.Carro;
import sistemaoficina.Pessoa.Pessoa;
import java.util.ArrayList;
import java.util.List;

public class Cliente extends Pessoa{
    private List<Carro> carros;
    
    //Construtor
    public Cliente(String nome, String cpf, String rg, String endereco, String telefone, String email) {
        super(nome, cpf, rg, endereco, telefone, email);
        carros = new ArrayList<>();
    }

    public Cliente() {}

    public void setAtributos(Cliente envia) {
        this.setNome(envia.getNome());
        this.setCpf(envia.getCpf());
        this.setRg(envia.getRg());
        this.setEndereco(envia.getEndereco());
        this.setTelefone(envia.getTelefone());
        this.setEmail(envia.getEmail());
    }
    
    //Interacaoes com carro

    //Adicionar carro
    public void addCarro(Carro carro) {
        if(!carros.isEmpty())
            carros.add(carro);
    }
    
    //Remover Carro
    protected void popCarro(Carro carro) {
        if(!carros.isEmpty())
            carros.remove(carros);
    }
    
    //Comparação
    @Override
    public boolean equals(Object o) {
        if(this == o)
            return true;
        
        if(o == null)
            return false;
        
        if(getClass() != o.getClass())
            return false;

        Cliente cliente = (Cliente)o;
        
        return ((this.getCpf()).equals(cliente.getCpf()) && (this.getRg()).equals(cliente.getRg()));
    }
}